pkg/apis/pipeline/v1beta1/resultref.go

```diff
@@ -52,7 +52,9 @@ const (
 
 var variableSubstitutionRegex = regexp.MustCompile(variableSubstitutionFormat)
 var resultNameFormatRegex = regexp.MustCompile(ResultNameFormat)
-var arrayIndexingRegex = regexp.MustCompile(arrayIndexing)
+
+// ArrayIndexingRegex is used to match `[int]` and `[*]`
+var ArrayIndexingRegex = regexp.MustCompile(arrayIndexing)
```

Comment:
I'm wondering if we need to export this regex or if there is a parsing function it would make more sense to export

Reply:
oh that's a good point! Let me check if I can have a separate function for this

Reply:
could you please avoid exporting the regexes here?